### ISSUE TITLE
Test assert for size_t and enums?

### ISSUE THREAD

**Issue by wolf99** [ThrowTheSwitch/Unity#437]
Is there an assert suggested for testing values of type `size_t`?

**Comment by wolf99** [comment]
Similar also goes for `enum`. Both types are of implementation defined size.

**Comment by mvandervoord** [comment]
I've added a macro for size_t. There's not a universal way to do enums because some optimizing compilers treat the following two enums as different sizes:

```
typedef enum _ENUM1_T
{
   FIRST = 0,
   SECOND = 1
} ENUM1_T;

typedef enum _ENUM2_T
{
   FIRST = 0,
   SECOND = 500
} ENUM2_T;
```

We've come across compilers that optimize the first to be a UINT8 and the second to a UINT16. Other compilers will make them both UIN32 or UINT64's. So there's not a lot of consistency we can count on. 

We COULD cast them all to UINT which will use the largest unsigned int which will always be big enough. This works well for everything except arrays, and it's how I implemented the size_t... but for an enum it seems it's best to leave it to the test writer to say which macro they want to display the information as.
